src/main/java/ch/njol/skript/expressions/ExprPersistentData.java

```diff
@@ -0,0 +1,289 @@
+		for (Expression<?> expr : variables.getExpressions()) {
+			String varName = ((Variable<?>) expr).getName().toString(e);
+			if (varName.contains(Variable.SEPARATOR)) { // It's a list
+				for (Object holder : holders.getArray(e)) {
+					for (Object object : PersistentDataUtils.getList(holder, varName))
+						values.add(object);
+				}
+			} else { // It's a single variable
+				for (Object holder : holders.getArray(e))
+					values.add(PersistentDataUtils.getSingle(holder, varName));
+			}
+		}
+		try {
+			return Converters.convertStrictly(values.toArray(), superType);
+		} catch (ClassCastException e1) {
+			return (T[]) Array.newInstance(superType, 0);
+		}
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.RESET)
+			return null;
+		for (Expression<?> expr : variables.getExpressions()) {
+			if (!((Variable<?>) expr).isList()) {
+				if (mode == ChangeMode.REMOVE_ALL)
+					return null;
+				return CollectionUtils.array(Object.class);
+			}
+		}
+		return CollectionUtils.array(Object[].class);
+	}
+
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta == null && mode != ChangeMode.DELETE)
+			return;
+		switch (mode) {
+			case SET:
+				for (Expression<?> expr : variables.getExpressions()) {
+					Variable<?> var = (Variable<?>) expr;
+					String varName = var.getName().toString(e);
+					if (var.isList()) {
+						varName = varName.replace("*", "");
+						for (Object holder : holders.getArray(e)) {
+							for (int i = 1; i <= delta.length; i++) {
+								// varName + i = var::i (e.g. exampleList::1, exampleList::2, etc.)
+								PersistentDataUtils.setList(holder, varName + i, delta[i - 1]);
+							}
+						}
+					} else if (varName.contains(Variable.SEPARATOR)) { // Specific index of a list
+						for (Object holder : holders.getArray(e))
+							PersistentDataUtils.setList(holder, varName, delta[0]);
+					} else { // It's a single variable
+						for (Object holder : holders.getArray(e))
+							PersistentDataUtils.setSingle(holder, varName, delta[0]);
+					}
+				}
+				break;
+			case DELETE:
+				for (Expression<?> expr : variables.getExpressions()) {
+					String varName = ((Variable<?>) expr).getName().toString(e);
+					if (varName.contains(Variable.SEPARATOR)) { // It's a list
+						for (Object holder : holders.getArray(e))
+							PersistentDataUtils.removeList(holder, varName);
+					} else { // It's a single variable
+						for (Object holder : holders.getArray(e))
+							PersistentDataUtils.removeSingle(holder, varName);
+					}
+				}
+				break;
+			case ADD:
+				for (Expression<?> expr : variables.getExpressions()) {
+					Variable<?> var = (Variable<?>) expr;
+					String varName = var.getName().toString(e);
+					if (var.isList()) {
+						for (Object holder : holders.getArray(e)) {
+							Map<String, Object> varMap = PersistentDataUtils.getListMap(holder, varName);
+							if (varMap != null) {
```

Comment:
Note to myself: This section needs reworked. varName should have the "*" replaced before the holder loop and this should still work even if varMap is null